dist/jszip-utils.js

```diff
@@ -89,20 +100,35 @@ JSZipUtils.getBinaryContent = function(path, callback) {
                     err = null;
                     try {
                         file = JSZipUtils._getBinaryFromXHR(xhr);
-                    } catch(e) {
-                        err = new Error(e);
+                        cbo.done(file);
+                    } catch(rr) {
+                        err = new Error(rr);
+                        cbo.fail(err);
                     }
-                    callback(err, file);
                 } else {
-                    callback(new Error("Ajax error for " + path + " : " + this.status + " " + this.statusText), null);
+                    cbo.fail(new Error("Ajax error for " + path + " : " + this.status + " " + this.statusText));
                 }
             }
         };
+        
+        if(cbo.progress) {
+            xhr.onprogress = function(e) {
+                cbo.onprogress.call(xhr, {
+                    path: path,
+                    originalEvent: e,
+                    percent: e.loaded / e.total * 100,
+                    downloaded: e.loaded,
+                    size: e.total
+                });
```

Comment:
Exposing the `xhr` as `this` in the callback exposes the internals. If we wanted to switch to `fetch` in the future (once it supports progress events) that would be hard. I'd rather just call `options.progress()`

I think we should also just expose `loaded` and `total` from this event.

Reply:
I agree